libraries/Microsoft.Bot.Builder.Dialogs/Prompts/ActivityPrompt.cs

```diff
@@ -142,7 +152,11 @@ protected virtual async Task OnPromptAsync(ITurnContext turnContext, IDictionary
                 throw new ArgumentNullException(nameof(options));
             }
 
-            if (options.Prompt != null)
+            if (isRetry == true && options.RetryPrompt != null)
```

Comment:
Minor, isRetry == true is redundant, we can just probably use isRetry that is bool.
The tests don't seem to catch this change either, not sure if we are missing an Assert somewhere.